src/main/scala/dotty/dokka/model/api/api.scala

```diff
@@ -99,9 +99,28 @@ extension (s: Signature):
 
 case class LinkToType(signature: Signature, dri: DRI, kind: Kind)
 
-case class HierarchyDiagram(edges: Seq[Edge])
-case class Vertex(val id: Int, val body: LinkToType)
-case class Edge(val from: Vertex, val to: Vertex)
+case class Vertex(id: Int, body: LinkToType)
+case class Edge(from: Vertex, to: Vertex)
+case class HierarchyGraph private (edges: Seq[Edge], private val nextId: Int):
```

Comment:
Maybe we can store only Edge data (`(LinkToType, LinkToType)`) in the graph and compute Vertexes and Edges on demand? Then we do not need thsi complicated logic to store nextId and code will be much faster

Reply:
TBH we can store graph as just pairs and use that pairs to generate graph itself so probably we may not need at all:
```
case class Vertex(id: Int, body: LinkToType)
case class Edge(from: Vertex, to: Vertex)
```